### PR TITLE
Handle ActorInitializationException in supervisors Closes #1079

### LINKED ISSUE #1079: Workflow gets stuck in Initialization if BackendWorkflowInitializationActor crashe during instantiation
If `BackendWorkflowInitializationActor` fails to instantiate, it never receives the `Initialize` message and the WorkflowActor never gets a response. So the workflow hangs forever.

From a more general perspective, if any actor in the lifecycle craches/dies unexpectedly, the workfow will hang at whichever state it was in..

### PR DESCRIPTION
**Pull request by Horneth**

### CODE REVIEW

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesFinalizationActor.scala [hunk 1]
```diff
@@ -39,9 +38,7 @@ class JesFinalizationActor (override val workflowDescriptor: BackendWorkflowDesc
 
   private def deleteAuthenticationFile(): Future[Unit] = {
     if (jesConfiguration.needAuthFileUpload) {
-      val delete = () => Future(workflowPaths.gcsAuthFilePath.delete(false))
-
-      Retry.withRetry(delete, isFatal = isFatalJesException, isTransient = isTransientJesException)(context.system) map { _ => () }
+      Future(workflowPaths.gcsAuthFilePath.delete(false)) map { _ => () }
```
Comment (cjllanwarne) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesFinalizationActor.scala]: No more retry?

**Comment by cjllanwarne** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/broadinstitute/cromwell/pull-request/1084/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/cromwell)

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesFinalizationActor.scala (hunk 1 above)
Comment (Horneth) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesFinalizationActor.scala]: there's already a retry 3 times in the gcs filesystem

**Comment by mcovarr** [comment]
This seems to fix the specific issue that was causing the workflow to get stuck in #1079, but does it address the more general problem of the workflow getting stuck if the initialization/finalization actors fail to instantiate for some other reason?

**Comment by Horneth** [comment]
@mcovarr correct, the more general problem exists throughout cromwell though and I didn't want to get into a rabbit hole with this but after digging a bit more it's actually not that bad. I'll update the PR with some better crash handling instead

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/WorkflowLifecycleActor.scala [hunk 2]
```diff
@@ -65,6 +66,12 @@ trait WorkflowLifecycleActor[S <: WorkflowLifecycleActorState] extends LoggingFS
   def successResponse: WorkflowLifecycleSuccessResponse
   def failureResponse(reasons: Seq[Throwable]): WorkflowLifecycleFailureResponse
 
+  override def supervisorStrategy = OneForOneStrategy() {
+    case ex: ActorInitializationException =>
+      context.parent ! failureResponse(Seq(ex))
+      Stop
```
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/WorkflowLifecycleActor.scala]: This is a little confusing: 

I think we also want to stop this actor, and any other supervisees other than the one which failed?

Or will that happen naturally when the parent is stopped?

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/WorkflowLifecycleActor.scala [hunk 3]
```diff
@@ -65,6 +66,12 @@ trait WorkflowLifecycleActor[S <: WorkflowLifecycleActorState] extends LoggingFS
...
+    case t => super.supervisorStrategy.decider.applyOrElse(t, (_: Any) => Escalate)
```
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/WorkflowLifecycleActor.scala]: What does this get escalated to? I can't see a supervisor strategy in WorkflowActor?

Comment (Horneth) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/WorkflowLifecycleActor.scala]: Every actor has a default supervisor strategy http://doc.akka.io/docs/akka/2.4.7/scala/fault-tolerance.html#Default_Supervisor_Strategy

**Comment by mcovarr** [comment]
:+1: assuming you can convince Travis

**Comment by cjllanwarne** [comment]
re-👍, pending test passes

**Comment by cjllanwarne** [comment]
Going to merge. Centaur has been updated to address the test failures.

**Pull request merged**